beacon-chain/blockchain/process_block_helpers.go

```diff
@@ -209,35 +209,35 @@ func (s *Service) fillInForkChoiceMissingBlocks(ctx context.Context, blk interfa
 	return s.cfg.ForkChoiceStore.InsertChain(ctx, pendingNodes)
 }
 
-// inserts finalized deposits into our finalized deposit trie.
-func (s *Service) insertFinalizedDeposits(ctx context.Context, fRoot [32]byte) error {
-	ctx, span := trace.StartSpan(ctx, "blockChain.insertFinalizedDeposits")
+// inserts finalized deposits into our finalized deposit trie, needs to be
+// called in the background
+func (s *Service) insertFinalizedDeposits(fRoot [32]byte) {
+	ctx, span := trace.StartSpan(context.Background(), "blockChain.insertFinalizedDeposits")
 	defer span.End()
 
 	// Update deposit cache.
 	finalizedState, err := s.cfg.StateGen.StateByRoot(ctx, fRoot)
 	if err != nil {
-		return errors.Wrap(err, "could not fetch finalized state")
+		log.WithError(err).Error("could not fetch finalized state")
```

Comment:
these errors should not prevent block insertion

Reply:
They don't there are in an alternate routine for that purpose

Reply:
If we end up changing errors to logs, we should capitalize messages

Reply:
This is a log with an error, I treat it thus the same as an error message.

Reply:
But its not the same, the finalized state will be nil if there is an error and we will panic due to accessing a nil pointer.